src/Control/Retry.hs

```diff
@@ -96,27 +105,38 @@ import           Prelude                hiding (catch)
 -- For anything more complex, just define your own 'RetryPolicy':
 --
 -- >> myPolicy = RetryPolicy $ \ n -> if n > 10 then Just 1000 else Just 10000
-newtype RetryPolicy = RetryPolicy { getRetryPolicy :: Int -> Maybe Int }
+newtype RetryPolicyM m = RetryPolicyM { getRetryPolicyM :: Int -> m (Maybe Int) }
 
 
-instance Default RetryPolicy where
+-- | 'RetryPolicyM' without monadic logic in determining the reply.
+type RetryPolicy = RetryPolicyM Identity
```

Comment:
That works with most of the functions, but not `RetryPolicy` transformers. If we keep the type signature of `capDelay` from master, for example, the following wouldn't typecheck. Still, I agree this makes more sense than using Identity there.

```haskell
let a = fullJitterBackoff 5 :: RetryPolicyM IO

-- This'll give you a type error.
let b = let b = capDelay 10 a
```